### PR TITLE
Hospital model

### PR DESCRIPTION
**Pull request by Alfredbis29**
In this Milestone of our Project I did the following ones: 

- Created a new Hospital model with properties: `name`, `info`, `address`, `treatments_counter`, `healthcare_requests_counter`, and `user_id`
- Defined relationships with the Treatment and HealthcareRequest models
- Created a Hospitals controller with standard CRUD actions and a `before_action` to set the @hospital instance variable
- Defined private methods to handle strong parameters and set the @hospital instance variable
- Added validations for the Hospital model properties to ensure they are present and have valid numerical values

### CODE REVIEW

File: Procfile.dev [hunk 1]
```diff
@@ -1,2 +1,2 @@
-web: bin/rails server -p 3000
-css: bin/rails tailwindcss:watch
```
Comment (Tresor11) [inline Procfile.dev:2]: Why did you change this?

File: app/models/hospital.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Hospital < ApplicationRecord
+  belongs_to :user
```
Comment (Tresor11) [inline app/models/hospital.rb:2]: We do not need the associations here
You should just create the hospital model with a name and address attribute

File: config/database.yml [hunk 3]
```diff
@@ -18,17 +18,17 @@ default: &default
   adapter: postgresql
   encoding: unicode
 
-  # username: postgres
-  # password: alfred
+  username: postgres
+  password: alfred
```
Comment (Tresor11) [inline config/database.yml:22]: Remove this please

File: config/database.yml [hunk 4]
```diff
@@ -18,17 +18,17 @@ default: &default
...
   # https://guides.rubyonrails.org/configuring.html#database-pooling
   pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
 
 development:
   <<: *default
   database: health_care_online_system_development
-  # username: postgres
-  # password: alfred
+  username: postgres
+  password: alfred
```
Comment (Tresor11) [inline config/database.yml:31]: Remove this as well

File: Gemfile [hunk 5]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
-ruby '3.1.2'
+ruby '3.0.1'
```
Comment (Tresor11) [inline Gemfile:4]: Don't change the ruby version

File: Gemfile [hunk 6]
```diff
@@ -11,6 +11,8 @@ gem 'rails', '~> 7.0.4', '>= 7.0.4.3'
 # The original asset pipeline for Rails [https://github.com/rails/sprockets-rails]
 gem 'sprockets-rails'
 
+gem 'sidekiq'
```
Comment (Tresor11) [inline Gemfile:14]: Why re you setting up sidekiq?

**Review by Tresor11 — CHANGES_REQUESTED** [review]

File: app/controllers/hospitals_controller.rb [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  @hospital.destroy
+
+  redirect_to hospitals_path
+end
+
+private
+
+def hospital_params
+  params.require(:hospital).permit(:name, :treatments_counter, :healthcare_requests_counter, :data, :address, :status, :user_id)
+  end
```
Comment (BenMukebo) [inline app/controllers/hospitals_controller.rb:49]: - [ ] Kindly make sure to correctly set up the hospital_params for the Jsonb records. Columns which has jsonb type are expected to contain a different type of information (int, string, array, ...) you need to find all the necessary information related o the hospital.
- For example, the address of the hospital may have country, city, and state. .....

```ruby
{ address: %i[country city state zip_code] }
```

**Review by Tresor11 — APPROVED** [review]

File: app/controllers/hospitals_controller.rb (hunk 7 above)
Comment (BenMukebo) [reply app/controllers/hospitals_controller.rb]: Hello @Tresor11, As suggested here, I made some pushes. Please check them out and tell me what are missing ?

**Comment by BenMukebo** [comment]
@Tresor11, Still waiting for review here as weel !!

**Pull request merged**